[docs] `signatureSubscribe` websocket (#31841)
solana-labs/solana: docs/src/api/websocket/_signatureSubscribe.mdx

## signatureSubscribe

> Subscribe to a transaction signature, to receive notification when the given

Reviewer: ```suggestion
Subscribe to a transaction signature, to receive a notification when the given
```

Reviewer: Better:

> Subscribe to receive a notification when the transaction with the given signature reaches…

  href="/api/http#configuring-state-commitment"
></Field>

<Field name="enableReceivedNotification" type="bool" optional={true}>

> Whether or not to recieve notifications when signatures are recieved by the RPC,

Reviewer: ```suggestion
Whether or not to subscribe for notifications when signatures are received by the RPC,
```

#### Notification Format:

The notification will be an RpcResponse JSON object with value containing an
object with:

- `slot: <u64>` - The corresponding slot.
- `value: <object|string>` - notification response value of:
  - `err: <object|null>`: a

>     [`RpcSignatureResult`](https://github.com/solana-labs/solana/blob/6d28fd455b07e3557fc6c0c3ddf3ba03e3fe8482/rpc-client-api/src/response.rs#L265)

Reviewer: ```suggestion
    [`RpcSignatureResult`](https://github.com/solana-labs/solana/blob/6d28fd455b07e3557fc6c0c3ddf3ba03e3fe8482/rpc-client-api/src/response.rs#L265-L268)
```

    with a value of either:
    - `null` if transaction succeeded in being processed at the specified
      commitment level, or
    - a

>       [`TransactionError`](https://github.com/solana-labs/solana/blob/6d28fd455b07e3557fc6c0c3ddf3ba03e3fe8482/sdk/src/transaction/error.rs#L15),

Reviewer: ```suggestion
      [`TransactionError`](https://github.com/solana-labs/solana/blob/6d28fd455b07e3557fc6c0c3ddf3ba03e3fe8482/sdk/src/transaction/error.rs#L15-L164),
```

      if the transaction failed
  - when `enableReceivedNotification` is true,

>     [`ReceivedSignatureResult`](https://github.com/solana-labs/solana/blob/6d28fd455b07e3557fc6c0c3ddf3ba03e3fe8482/rpc-client-api/src/response.rs#L286)

Reviewer: ```suggestion
    [`ReceivedSignatureResult`](https://github.com/solana-labs/solana/blob/6d28fd455b07e3557fc6c0c3ddf3ba03e3fe8482/rpc-client-api/src/response.rs#L286-L288)
```
